File: cses/01-introductory/11-coin-piles/CoinPiles.java
import java.util.*;

public class CoinPiles {
    public static void main(String[] args) {
	    CoinPiles sol = new CoinPiles();
		Scanner sc = new Scanner(System.in);
		int t = sc.nextInt();

		while (t-- > 0) {
			long a = sc.nextLong();
			long b = sc.nextLong();

			long mx = Math.max(a, b);
			long mi = Math.min(a, b);

			long diff = mx - mi;

			mx -= 2 * diff;
			mi -= diff;
			StringBuilder sb = new StringBuilder();

			if (mi == 0 || (mi > 0 && mi % 3 == 0)) {
				sb.append("YES\n");
			} else {
				sb.append("NO\n");
			}

			System.out.println(sb);
		}
	}
	
    
}

File: cses/01-introductory/11-coin-piles/coinpiles.cpp
#include <bits/stdc++.h>

using namespace std;

const long long MOD = 1e9 + 7;
const long long INF = 1e9 + 7;

typedef long long ll;
typedef vector<int> vi;
typedef pair<int,int> ii;
typedef vector<ii> vii;
typedef vector<vi> vvi;

#define F first
#define S second
#define PB push_back
#define MP make_pair

void solve()
{
  ll a, b;
  cin >> a >> b;

  ll mx = max(a, b);
  ll mi = min(a, b);

  ll diff = mx - mi;

  mx -= 2 * diff;
  mi -= diff;

  if (mi == 0 || (mi > 0 && mi % 3 == 0)) {
	cout << "YES" << endl;
  } else {
	cout << "NO" << endl;
  }
  
}

int main() {
  ios::sync_with_stdio(0);
  cin.tie(0);

  int tc; cin >> tc;
  while (tc -- > 0) {
	  solve();
  }
}
